examples/validate-fee/modules/shared_data/src/main/scala/com/my/validate_fee/shared_data/types/Types.scala

```diff
@@ -0,0 +1,28 @@
+package com.my.validate_fee.shared_data.types
+
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import org.tessellation.currency.dataApplication.{DataCalculatedState, DataOnChainState, DataUpdate}
+import org.tessellation.currency.schema.feeTransaction.FeeTransaction
+import org.tessellation.schema.balance.Amount
+import eu.timepit.refined.auto._
+
+object Types {
+  @derive(decoder, encoder)
+  sealed trait SampleUpdate extends DataUpdate
+
+  val updateTypeOneFee = Amount(123L)
+  val updateTypeTwoFee = Amount(1000L)
```

Comment:
This example is good but maybe we could extend the functionality a bit. Can we make the type two fee variable in some way?

Like, maybe have the fee make use of calculated state to determine the amount based on usage? So the 1st request from a wallet would be a fee of 1, 2nd fee of 2, etc. 

Or else we could use snapshot ordinal floor 10, or something like that to make it variable: `fee = floor(ordinal/10)`

Open to ideas but it would be great if we could show a slightly more advanced fee functionality (like one of the above) alongside a basic static fee.

Reply:
@AlexBrandes Is calculated state persisted between node restarts? (This is metagraph-dependent, isn't it?) If not, won't that make fee calculations based on calculated state non-deterministic?

Reply:
Calculated state is recalculated after each restart so it's available in the same form/persisted.